opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/OpenSearchIndexScanQueryBuilder.java

```diff
@@ -99,6 +103,18 @@ public boolean pushDownHighlight(LogicalHighlight highlight) {
     return true;
   }
 
+  private boolean trackScoresFromOpenSearchFunction(Expression condition) {
+    if (condition instanceof OpenSearchFunctions.OpenSearchFunction) {
+      return ((OpenSearchFunctions.OpenSearchFunction) condition).isScoreTracked();
+    }
```

Comment:
This picks `isScoreTracked` from the first function seen. I think you need to accumulate the result using `||` from all functions.

Reply:
Thanks.  I thought that might be my bug.  The intention was to only return if .isScoreTracked() is true.  I'll update that.